tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetService.scala

```diff
@@ -0,0 +1,24 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import jakarta.inject.Inject
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.method.IdentityResolver
+import org.apache.james.mailbox.MailboxSession
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class PublicAssetSetService @Inject()(val identityResolver: IdentityResolver) {
+  def checkIdentityIdsExist(identityIds: Seq[IdentityId], session: MailboxSession): SMono[Seq[IdentityId]] =
+    SFlux.fromIterable(identityIds)
+      .concatMap(identityId => checkIdentityIdExist(identityId, session))
+      .collectSeq()
```

Comment:
`IdentityResolver.resolveIdentityId` lists all Identities of a user.

Maybe we should not rely directly on `IdentityResolver` but wrap `IdentityRepository` to avoid listing all identities multiple times.

Reply:
Can you explain more?
I don't get it
![image](https://github.com/linagora/tmail-backend/assets/81145350/d98bce9b-31d7-41d6-b510-759233bcf056)

Reply:
If we use `IdentityResolver` to check if 3 ids exist for example, we list all the identities 3 times.

IMO we can just list all identities 1 time to check if 3 ids exist.

Reply:
Do you mean we should have a method
`IdentityRepository.listByUserNameAndIdentityIds(username: Username, identityids: Seq[IdentityId])` ?
or add `findByIdentityId(user: Username, identityId: IdentityId)` ?

I don't think we get more benefit from it
One user doesn't have a lot of identityIds (in real words)
The fetch-all and filter I think no big deal

Reply:
> One user doesn't have a lot of identityIds (in real words)
The fetch-all and filter I think no big deal

I mean still fetch-all and filter, but for just 1 times. 
e.g. inject `IdentityRepository` into `PublicAssetSetService`:
```scala
def checkIdentityIdsExist(identityIds: Seq[IdentityId], session: MailboxSession): SMono[Seq[IdentityId]] =
    identityRepository.list(session.getUser)
      .filter(identity => identityIds.contains(identity))
      .collectSeq()
```

Reply:
ah, good point